modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -71,11 +72,13 @@ func (k Keeper) OnChanOpenTry(
 	if order != channeltypes.ORDERED {
 		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "invalid channel ordering: %s, expected %s", order.String(), channeltypes.ORDERED.String())
 	}
-	if version != types.Version {
-		return sdkerrors.Wrapf(types.ErrInvalidVersion, "got: %s, expected %s", version, types.Version)
+
+	if err := types.ValidateVersion(version); err != nil {
+		return err
 	}
-	if counterpartyVersion != types.Version {
-		return sdkerrors.Wrapf(types.ErrInvalidVersion, "invalid counterparty version: %s, expected %s", counterpartyVersion, types.Version)
+
+	if err := types.ValidateVersion(counterpartyVersion); err != nil {
+		return err
```

Comment:
This error message should probably indicate that the `counterpartyVersion` failed, same with regular version above. Otherwise we won't be able to distinguish which version was incorrect from a quick glance

Reply:
Done!